libmscore/repeatlist.cpp

```diff
@@ -399,24 +399,29 @@ void RepeatList::unwind()
 
 Measure* RepeatList::jumpToStartRepeat(Measure* m)
       {
-      Measure* nm;
-      //
-      // go back to start of repeat or section break
-      // handle special case of end repeat (Measure m) has a section break
-      //
-      for (nm = m; nm && nm != _score->firstMeasure(); nm = nm->prevMeasure()) {
-            if (nm->repeatFlags() & Repeat::START || (nm->sectionBreak() && m != nm)) {
-                  if (nm->sectionBreak() && nm->nextMeasure())
-                        nm = nm->nextMeasure();
-                  break;
-                  }
-            }
+      // finalize the previous repeat segment
```

Comment:
I moved the finalize of previous repeat segment to beginning of function, because it makes more sense to deal with it at the start, and also so I don't have to keep track of two measure variables in this function.